pkg/operator/controller/gateway-service-dns/controller_test.go

```diff
@@ -96,79 +118,96 @@ func Test_Reconcile(t *testing.T) {
 			},
 			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{},
 		},
 		{
 			name: "gateway with two listeners and one dnsrecord with a stale target",
 			existingObjects: []runtime.Object{
+				dnsConfig, infraConfig,
 				gw(
 					"example-gateway",
 					l("http", "*.example.com", 80),
 					l("https", "*.example.com", 443),
 				),
-				svc(
-					"example-gateway",
-					map[string]string{
-						"gateway.istio.io/managed": "example-gateway",
-					},
-					map[string]string{
-						"istio.io/gateway-name": "example-gateway",
-					},
-					ingHost("newlb.example.com"),
-				),
-				dnsrecord("example-gateway-55bcfdb97d-wildcard", "*.example.com", "oldlb.example.com"),
+				svc("example-gateway", gatewayManagedLabel, exampleGatewayLabel, ingHost("newlb.example.com")),
+				dnsrecord("example-gateway-55bcfdb97d-wildcard", "*.example.com", iov1.ManagedDNS, exampleGatewayLabel, "oldlb.example.com"),
 			},
 			reconcileRequest: req("openshift-ingress", "example-gateway"),
 			expectCreate:     []client.Object{},
 			expectUpdate: []client.Object{
-				dnsrecord("example-gateway-55bcfdb97d-wildcard", "*.example.com", "newlb.example.com"),
+				dnsrecord("example-gateway-55bcfdb97d-wildcard", "*.example.com", iov1.ManagedDNS, exampleGatewayLabel, "newlb.example.com"),
+			},
+			expectDelete: []client.Object{},
+		},
+		{
+			name: "gateway with a stale dnsrecord",
+			existingObjects: []runtime.Object{
+				dnsConfig, infraConfig,
+				gw(
+					"example-gateway",
+					l("http", "*.new.example.com", 80),
+				),
+				svc("example-gateway", gatewayManagedLabel, exampleGatewayLabel, ingHost("lb.example.com")),
+				dnsrecord("example-gateway-57b76476b6-wildcard", "*.old.example.com", iov1.ManagedDNS, exampleGatewayLabel, "lb.example.com"),
+			},
+			reconcileRequest: req("openshift-ingress", "example-gateway"),
+			expectCreate: []client.Object{
+				dnsrecord("example-gateway-6d59bf9b46-wildcard", "*.new.example.com", iov1.ManagedDNS, exampleGatewayLabel, "lb.example.com"),
+			},
+			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{
+				dnsrecord("example-gateway-57b76476b6-wildcard", "*.old.example.com", iov1.ManagedDNS, exampleGatewayLabel, "lb.example.com"),
 			},
 		},
+		{
+			name: "gateway with a listener with an unmanaged domain, no dnsrecords",
+			existingObjects: []runtime.Object{
+				dnsConfig, infraConfig,
+				gw(
+					"example-gateway",
+					l("http", "*.foo.com", 80),
+				),
+				svc("example-gateway", gatewayManagedLabel, exampleGatewayLabel, ingHost("lb.example.com")),
+			},
+			reconcileRequest: req("openshift-ingress", "example-gateway"),
+			expectCreate: []client.Object{
+				dnsrecord("example-gateway-957dccf84-wildcard", "*.foo.com", iov1.UnmanagedDNS, exampleGatewayLabel, "lb.example.com"),
+			},
+			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{},
+		},
```

Comment:
Is it worth adding 2 simple test cases: one for `dnsConfig` is null and another for `infraConfig` is null and nothing should be created?

Reply:
Done in https://github.com/openshift/cluster-ingress-operator/compare/038190ece2d957a26615bbc666ad9164a2e67846..0eb49fbfeccdb763ad5757025967393ac0f1465c.